### PR TITLE
Copy edit - Guides: Ecto

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: guides/ecto.md [hunk 1]
```diff
@@ -2,23 +2,27 @@
...
+Most web applications today need some form of data validation and persistence. In the Elixir ecosystem, we have [`Ecto`](https://hexdocs.pm/ecto/) to enable this. Before we jump into building database-backed web features, we're going to focus on the finer details of Ecto to give a solid base to build our web features on top of. Let's get started!
 
 Phoenix uses Ecto to provide builtin support to the following databases:
 
-* PostgreSQL (via [`postgrex`](https://github.com/elixir-ecto/postgrex))
-* MySQL (via [`myxql`](https://github.com/elixir-ecto/myxql))
-* MSSQL (via [`tds`](https://github.com/livehelpnow/tds))
+* PostgreSQL (via [`postgrex`](https://hexdocs.pm/postgrex/))
+* MySQL (via [`myxql`](https://hexdocs.pm/myxql/))
+* MSSQL (via [`tds`](https://hexdocs.pm/tds/))
```
Comment (eksperimental) [inline guides/ecto.md:11]: Here I am not 100% convinced it is best to link to the docs.

File: guides/ecto.md [hunk 2]
```diff
@@ -2,23 +2,27 @@
...
+Ecto also provides support for other databases and it has many learning resources available. Please check out [Ecto's README file](https://github.com/elixir-ecto/ecto/blob/master/README.md) for general information.
 
 This guide assumes that we have generated our new application with Ecto integration and that we will be using PostgreSQL. The introductory guides cover how to get your first application up and running. For instructions on switching to MySQL, please see the [Using MySQL](#using-mysql) section.
 
 ## Using the schema and migration generator
 
-Now that we have Ecto and Postgres installed and configured, the easiest way to use Ecto is to generate an Ecto *schema* through the `phx.gen.schema` task. Ecto schemas are a way for us to specify how Elixir data types map to and from external sources, such as database tables. Let's generate a `User` schema with `name`, `email`, `bio`, and `number_of_pets` fields.
+[NOTE]: # (It will also be good to show two commands.  one to see if Posgresql is installed, and the minimum required version and another one for Ecto)
```
Comment (eksperimental) [inline guides/ecto.md:23]: Improvement.

File: guides/ecto.md [hunk 3]
```diff
@@ -122,11 +125,11 @@ defmodule Hello.Repo do
 end
 ```
 
-It begins by configuring our `otp_app` name and repo module. Then it sets the adapter – Postgres, in our case. It also sets our login credentials. Of course, you can change these to match your actual credentials if they are different.
```
Comment (eksperimental) [inline guides/ecto.md:125]: I deleted the second sentence as nowhere in the code above it shows the credentials,
and it is more than obvious that can be edited in the code below. IMHO it adds no value.

File: guides/ecto.md [hunk 4]
```diff
@@ -2,23 +2,27 @@
 
 > **Requirement**: This guide expects that you have gone through the [introductory guides](installation.html) and got a Phoenix application [up and running](up_and_running.html).
 
-Most web applications today need some form of data validation and persistence. In the Elixir ecosystem, we have Ecto to enable this. Before we jump into building database-backed web features, we're going to focus on the finer details of Ecto to give a solid base to build our web features on top of. Let's get started!
+Most web applications today need some form of data validation and persistence. In the Elixir ecosystem, we have [`Ecto`](https://hexdocs.pm/ecto/) to enable this. Before we jump into building database-backed web features, we're going to focus on the finer details of Ecto to give a solid base to build our web features on top of. Let's get started!
```
Comment (josevalim) [inline guides/ecto.md:5]:
```suggestion
Most web applications today need some form of data validation and persistence. In the Elixir ecosystem, we have `Ecto` to enable this. Before we jump into building database-backed web features, we're going to focus on the finer details of Ecto to give a solid base to build our web features on top of. Let's get started!
```

File: guides/ecto.md (hunk 1 above)
Comment (josevalim) [reply guides/ecto.md:11]: It is not, The github projects is usually the ones with installation instructions. Please revert. :)

File: guides/ecto.md [hunk 5]
```diff
@@ -2,23 +2,27 @@
...
+* PostgreSQL (via [`postgrex`](https://hexdocs.pm/postgrex/))
+* MySQL (via [`myxql`](https://hexdocs.pm/myxql/))
+* MSSQL (via [`tds`](https://hexdocs.pm/tds/))
+* ETS (via https://github.com/evadne/etso [`etso`](https://hexdocs.pm/etso/))
+
 
 Newly generated Phoenix projects include Ecto with the PostgreSQL adapter by default. You can pass the `--database` option to change or `--no-ecto` flag to exclude this.
 
-Ecto also provides support for other databases and it has many learning resources available. Please check out [Ecto's README](https://github.com/elixir-ecto/ecto) for general information.
+Ecto also provides support for other databases and it has many learning resources available. Please check out [Ecto's README file](https://github.com/elixir-ecto/ecto/blob/master/README.md) for general information.
```
Comment (josevalim) [inline guides/ecto.md:17]: Please revert this change, linknig to the project root is much better than the README specific file:

```suggestion
Ecto also provides support for other databases and it has many learning resources available. Please check out [Ecto's README](https://github.com/elixir-ecto/ecto) for general information.
```

File: guides/ecto.md (hunk 2 above)
Comment (josevalim) [reply guides/ecto.md:23]: Feel free to send a separate PR to discuss this, probably doesn't belong to something doing copy edits.

File: guides/ecto.md [hunk 6]
```diff
@@ -543,6 +580,23 @@ $ mix ecto.migrate
...
+[`Ecto.Repo`]: https://hexdocs.pm/ecto/Ecto.Repo.html
+[`Repo.insert/1`]: https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert/2
+[`Repo.one/1`]: https://hexdocs.pm/ecto/Ecto.Repo.html#c:one/2
+[`Repo.update/2`]: https://hexdocs.pm/ecto/Ecto.Repo.html#c:update/2
+[`Repo.delete/2`]: https://hexdocs.pm/ecto/Ecto.Repo.html#c:delete/2
+[`Repo.insert_all/3`]: https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert_all/3
+[`Repo.update_all/3`]: https://hexdocs.pm/ecto/Ecto.Repo.html#c:update_all/3
+[`Repo.delete_all/2`]: https://hexdocs.pm/ecto/Ecto.Repo.html#c:delete_all/2
+[`Ecto.Query`]: https://hexdocs.pm/ecto/Ecto.Query.html
+[`from/2`]: https://hexdocs.pm/ecto/Ecto.Query.html#from/2
```
Comment (josevalim) [inline guides/ecto.md:602]: Please revert this stule, we don't use it anywhere else in the guides. Plus we should not link ``Ecto.Changeset.cast/3``, just let ex_doc do its job. Thank you.

Comment (eksperimental) [reply guides/ecto.md]: done

**Comment by eksperimental** [comment]
@josevalim All suggestions have been implemented

File: guides/ecto.md [hunk 7]
```diff
@@ -2,23 +2,25 @@
 
 > **Requirement**: This guide expects that you have gone through the [introductory guides](installation.html) and got a Phoenix application [up and running](up_and_running.html).
 
-Most web applications today need some form of data validation and persistence. In the Elixir ecosystem, we have Ecto to enable this. Before we jump into building database-backed web features, we're going to focus on the finer details of Ecto to give a solid base to build our web features on top of. Let's get started!
+Most web applications today need some form of data validation and persistence. In the Elixir ecosystem, we have [`Ecto`](https://hexdocs.pm/ecto/) to enable this. Before we jump into building database-backed web features, we're going to focus on the finer details of Ecto to give a solid base to build our web features on top of. Let's get started!
```
Comment (josevalim) [inline guides/ecto.md:5]:
```suggestion
Most web applications today need some form of data validation and persistence. In the Elixir ecosystem, we have `Ecto` to enable this. Before we jump into building database-backed web features, we're going to focus on the finer details of Ecto to give a solid base to build our web features on top of. Let's get started!
```

File: guides/ecto.md [hunk 8]
```diff
@@ -2,23 +2,25 @@
...
 * PostgreSQL (via [`postgrex`](https://github.com/elixir-ecto/postgrex))
 * MySQL (via [`myxql`](https://github.com/elixir-ecto/myxql))
 * MSSQL (via [`tds`](https://github.com/livehelpnow/tds))
+* ETS (via [`etso`](https://github.com/evadne/etso))
+
 
 Newly generated Phoenix projects include Ecto with the PostgreSQL adapter by default. You can pass the `--database` option to change or `--no-ecto` flag to exclude this.
 
-Ecto also provides support for other databases and it has many learning resources available. Please check out [Ecto's README](https://github.com/elixir-ecto/ecto) for general information.
+Ecto also provides support for other databases and it has many learning resources available. Please check out [Ecto's README file](https://github.com/elixir-ecto/ecto) for general information.
```
Comment (josevalim) [inline guides/ecto.md:17]:
```suggestion
Ecto also provides support for other databases and it has many learning resources available. Please check out [Ecto's README](https://github.com/elixir-ecto/ecto) for general information.
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:
